### PR TITLE
refactor: remove `FlushStatus` from channel end

### PR DESCRIPTION
**Pull request by damiannolan**
## Description

closes: #4278 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by DimitrisJim — APPROVED** [review]
finding some references in comments if you'd like to axe them: `grep -inr "flush status" --include="*.go"`

other than that, looks delicious

**Comment by damiannolan** [comment]
> grep -inr "flush status" --include="*.go"

Legend

**Review by colin-axner — APPROVED** [review]

File: modules/core/04-channel/types/channel.go [hunk 1]
```diff
@@ -25,8 +25,6 @@ func NewChannel(
 		Version:        version,
 		// UpgradeSequence is intentionally left empty as a new channel has not performed an upgrade.
 		UpgradeSequence: 0,
-		// FlushStatus is NOTINFLUSH as a new channel will not be in the upgrade process.
-		FlushStatus: NOTINFLUSH,
```
Comment (chatton) [inline modules/core/04-channel/types/channel.go:29]: 🪓

**Review by chatton — APPROVED** [review]
LGTM!

**Pull request merged**